File: src/main/java/io/gittul/domain/post/PostRepository.java
package io.gittul.domain.post;

import io.gittul.domain.post.entity.Post;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface PostRepository extends JpaRepository<Post, Long> {
    Page<Post> findAllByUserUserIdInOrderByCreatedAtDesc(List<Long> userIds, Pageable pageable);

    Page<Post> findAllByOrderByCreatedAtDesc(Pageable pageable);
}

File: app/src/main/kotlin/io/gittul/app/domain/post/PostRepository.kt
package io.gittul.app.domain.post

import io.gittul.core.domain.post.entity.Post
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface PostRepository : JpaRepository<Post, Long> {
    fun findAllByUserUserIdInOrderByCreatedAtDesc(userIds: MutableList<Long>, pageable: Pageable): Page<Post>

    fun findAllByOrderByCreatedAtDesc(pageable: Pageable): Page<Post>
}
